src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/builder/BigQueryWindowsAggregationSQLBuilder.java

```diff
@@ -0,0 +1,151 @@
+import java.util.stream.Collectors;
+
+/**
+ * Helper class used to generate BigQuery SQL Statements for Select statements.
+ */
+public class BigQueryWindowsAggregationSQLBuilder extends BigQueryBaseSQLBuilder {
+  private static final String ROW_NUM_PREFIX = "rn_";
+  private final WindowAggregationDefinition windowAggregationDefinition;
+  private static final Logger LOG = LoggerFactory.getLogger(BigQueryWindowsAggregationSQLBuilder.class);
+  private final String source;
+  private final String sourceAlias;
+  private final String rowNumColumnAlias;
+  private final StringBuilder builder;
+
+  public BigQueryWindowsAggregationSQLBuilder(WindowAggregationDefinition windowAggregationDefinition,
+                                              String sourceExpression,
+                                              String sourceAlias) {
+    this(windowAggregationDefinition,
+         sourceExpression,
+         sourceAlias,
+         ROW_NUM_PREFIX + BigQuerySQLEngineUtils.newIdentifier());
+
+  }
+
+  protected BigQueryWindowsAggregationSQLBuilder(WindowAggregationDefinition windowAggregationDefinition,
+                                                 String source,
+                                                 String sourceAlias,
+                                                 String rowNumColumnAlias) {
+    this.windowAggregationDefinition = windowAggregationDefinition;
+    this.source = source;
+    this.sourceAlias = sourceAlias;
+    // This is the alias used to store the row number value. Format is "rn_<uuid>"
+    this.rowNumColumnAlias = rowNumColumnAlias;
+    this.builder = new StringBuilder();
+  }
+
+  @VisibleForTesting
+  protected String getSelectedFields(Map<String, Expression> selectedFields) {
+    List<Expression> e = new ArrayList<>(selectedFields.values());
+    return getPartitionFields(e);
+  }
+
+  @Override
+  public String getQuery() {
+    String overClause = getOverClause();
+    String aggregateFields = getAggregateFields(overClause);
+    String query = SELECT + getSelectedFields(windowAggregationDefinition.getSelectExpressions()) + aggregateFields +
+      SPACE + FROM + OPEN_GROUP + SPACE + source + SPACE + CLOSE_GROUP + SPACE + AS + sourceAlias;
+    LOG.debug("Query is " + query);
+    return query;
+  }
+
+  @VisibleForTesting
+  protected String getAggregateFields(String overClause) {
+    String aggregateString = "";
+    Map<String, Expression> aggregateExpressions = windowAggregationDefinition.getAggregateExpressions();
+    for (String s : aggregateExpressions.keySet()) {
+      SQLExpression e = (SQLExpression) aggregateExpressions.get(s);
+      if (!"".equals(aggregateString)) {
+        aggregateString = aggregateString + COMMA;
+      }
+        aggregateString = aggregateString + SPACE + e.extract() + SPACE + overClause + AS + s;
+
+    }
+    return aggregateString;
+  }
+  
+  private String getOverClause() {
+    return OVER +
+      OPEN_GROUP + SPACE + PARTITION_BY + SPACE +
```

Comment:
add comments here showing how the expression starts to look like. 

See https://github.com/data-integrations/google-cloud/blob/645fd8056e2bd1e2fb5823725ef988318a902b27/src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/builder/BigQueryJoinSQLBuilder.java#L268-L275 as an example

Reply:
Done